File: src/main/Models/Buildable/Material/Nail.java
package main.Models.Buildable.Material;

/**
 * Nails used as material
 */
public enum Nail implements Material{
    TEN_D("10d");

    private final String niceString;

    Nail(String niceString) {
        this.niceString = niceString;
    }

    /**
     *
     * @return a nice String representation of the Nail
     */
    @Override
    public String toString() {
        return this.niceString + " nails";
    }
}

File: src/main/kotlin/models/buildable/Material/Nail.kt
package models.buildable.material

/**
 * Nails used as material
 */
enum class Nail(private val niceString: String) : Material {
    TEN_D("10d");

    /**
     *
     * @return a nice String representation of the Nail
     */
    override fun toString(): String {
        return "$niceString nails"
    }
}